src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Nullable.cs

```diff
@@ -2322,6 +2322,49 @@ class C
                 });
         }
 
+        [Fact]
+        public void EmitAttribute_UnconstrainedTypeParameter()
+        {
+            var source =
+@"#nullable enable
+public class Program
+{
+    public T F1<T>() => default!;
+    public T? F2<T>() => default;
+    public T F3<T>() where T : class => default!;
+    public T? F4<T>() where T : class => default;
+    public T F5<T>() where T : class? => default!;
+    public T F6<T>() where T : struct => default;
+    public T? F7<T>() where T : struct => default;
+    public T F8<T>() where T : notnull => default;
+    public T? F9<T>() where T : notnull => default!;
```

Comment:
❔ Was this supposed to be an option as well?

```suggestion
    public T? F9<T>() where T : notnull => default!;
    public T F10<T>() where T : default => default;
```

Reply:
`where T : default` can only be used for methods that override a virtual base method, to allow disambiguating between unconstrained `T?` and `Nullable<T>`, in the same way that `where T : class` or `where T : struct` is allowed for method overrides in C#8.

---
In reply to: [455541496](https://github.com/dotnet/roslyn/pull/45993#discussion_r455541496) [](ancestors = 455541496)